app/controllers/middleware_deployment_controller.rb

```diff
@@ -17,4 +28,48 @@ def show
     show_container(@record, controller_name, display_name)
   end
 
+  def button
+    selected_operation = params[:pressed].to_sym
+    if OPERATIONS.key?(selected_operation)
```

Comment:
This code block is (almost) the same as for the server controller - can we try to move it into a generic helper that gets the `OPERATIONS` passed in as an argument from the calling controller?